src/cargo/ops/fix.rs

```diff
@@ -154,6 +154,14 @@ fn check_version_control(config: &Config, opts: &FixOptions) -> CargoResult<()>
     if let Ok(repo) = git2::Repository::discover(config.cwd()) {
         let mut repo_opts = git2::StatusOptions::new();
         repo_opts.include_ignored(false);
+        if repo.is_empty()? && !opts.allow_dirty {
+            bail!(
+                "no commits found in the git repository, and \
```

Comment:
I feel this error is implementation centric rather than user centric. The core of the problem is not lack of commits but that there are uncommitted changes. We could just reuse the message below.

Unless I'm missing something, isn't it strange that git2 isn't reporting these and we have to add this?

Reply:
Good catch! I did some tests in https://github.com/rust-lang/cargo/issues/11380#issuecomment-1324423690

I think this behavior makes sense. Maybe we do not need to fix it? Or do we need to find all untracked files to warn it?

Reply:
Include untracked files. See https://github.com/rust-lang/cargo/issues/11380#issuecomment-1327557751